version: '3.8'

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    depends_on:
      db:
        condition: service_healthy
    restart: always
    environment:
      MYSQL_DATABASE: flaskdb
      MYSQL_HOST: db
      MYSQL_PASSWORD: flaskpassword
      MYSQL_USER: flaskuser
    ports:
      - "5000:5000"
    volumes:
      - .:/app
    networks:
      - app-network

  db:
    image: mysql:8.0
    environment:
      MYSQL_DATABASE: flaskdb
      MYSQL_PASSWORD: flaskpassword
      MYSQL_ROOT_PASSWORD: rootpassword
      MYSQL_USER: flaskuser
    ports:
      - "3306:3306"
    volumes:
      - db-data:/var/lib/mysql
    networks:
      - app-network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      retries: 5

networks:
  app-network:
    driver: bridge

volumes:
  db-data:
